conjur/resource.py

```diff
@@ -4,26 +4,49 @@
 Resource module
 """
 
+
 # pylint: disable=too-few-public-methods
+from conjur.errors import MissingRequiredParameterException
+
+
 class Resource:
     """
-    Used for representing Conjur resources
+    DTO class that represents a resource in Conjur.
     """
-    def __init__(self, type_:type, name:str):
-        self.type = type_
-        self.name = name
+
+    @classmethod
+    def from_full_id(cls, full_id: str):
+        """
+        Factory method for
+        """
+        id_parts = full_id.split(':', 2)
+        if len(id_parts) == 3:
+            # If identifier contains also the account part, remove it.
+            id_parts.pop(0)
+        elif len(id_parts) != 2:
+            raise MissingRequiredParameterException(
+                f"Resource id is missing 'kind:' prefix: {full_id}")
```

Comment:
@shulifink @szamir1 Please review text

Reply:
Let's discuss

Reply:
Resource ID missing 'kind:' prefix:

Reply:
Or "The resource ID is missing 'kind:' prefix:"